web/src/beta/lib/reearth-ui/components/TextInput/index.tsx

```diff
@@ -17,42 +19,52 @@ export const TextInput: FC<TextInputProps> = ({
   placeholder,
   size = "normal",
   disabled,
+  appearance,
   actions,
+  isEditable,
   onChange,
   onBlur,
 }) => {
   const [currentValue, setCurrentValue] = useState(value ?? "");
+  const [isFocused, setIsFocused] = useState(false);
 
   useEffect(() => {
     setCurrentValue(value ?? "");
   }, [value]);
 
-  const handleChange = useCallback(
-    (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
-      const newValue = e.currentTarget.value;
-      setCurrentValue(newValue ?? "");
-      onChange?.(newValue);
-    },
-    [onChange],
-  );
+  const handleChange = useCallback((e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    const newValue = e.currentTarget.value;
+    if (newValue === undefined) return;
+    setCurrentValue(newValue ?? "");
+  }, []);
 
   const handleBlur = useCallback(() => {
+    setIsFocused(false);
     onChange?.(currentValue);
```

Comment:
As an atom input, i think onChange should be triggered when change, and onBlur should be triggered when blur. We don't need onChange inside handleBlur and we need a onChange inside handleChange.